### PR TITLE
Proposals parity

### PR DESCRIPTION
**Pull request by motechFR**
### WHAT
copilot:summary

### WHY
Migrate proposals engine to the API

### CODE REVIEW

File: src/lib/permissions/index.ts [hunk 1]
```diff
@@ -2,4 +2,5 @@ export * from './interfaces';
 export * from './hasAccessToSpace';
 export * from './clients';
 export * from './forums';
+export * from './proposals';
```
Comment (mattcasey) [inline src/lib/permissions/index.ts:5]: each of these should probably be namespaced as well

File: src/lib/permissions/proposals/policies/__tests__/policyStatusVoteClosedViewOnly.spec.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
+import type { ProposalCategory, Space, User } from '@prisma/client';
+
+import type { ProposalWithUsers } from '../../../../proposals/interface';
+import { generateProposal, generateProposalCategory } from '../../../../testing/proposals';
```
Comment (mattcasey) [inline src/lib/permissions/proposals/policies/__tests__/policyStatusVoteClosedViewOnly.spec.ts:4]: is there a reason you are using relative paths here?

File: src/lib/proposals/interface.ts
Comment (mattcasey) [inline src/lib/proposals/interface.ts:1]: interfaces.ts seems to be our standard convention

File: src/lib/permissions/index.ts (hunk 1 above)
Comment (motechFR) [reply src/lib/permissions/index.ts:5]: Saw the PR in core, I like that pattern alot. I'd be +1 to have less top-level core and more specific exports, kinda like how we use Mui Material subpaths (without that extreme of granularity)

added @ charmverse/core/tests here for our utils

File: src/shared.ts [hunk 3]
```diff
@@ -5,6 +5,8 @@ export * from './lib/utilities/index';
 export * from './lib/permissions/clients/index';
 export * from './lib/permissions/interfaces';
 
+export * from './lib/proposals/interfaces';
```
Comment (mattcasey) [inline src/shared.ts:8]: Why do interfaces need to be exported from shared? they should be importable on the frontend using `type` just fine

**Pull request merged**